File: 3-binary_linear_classification/binary_linear_classification.jl
using Plots

default(framestyle=:box,grid=false,legend=false)

m = -0.3
b = 5
linear_function(x) = m*x.+b

x=range(0,10,length=100)
y=linear_function.(x)

above_scatter_y = @. linear_function.(x)+abs(0.9*randn())
below_scatter_y = @. linear_function.(x)-abs(0.6*randn())

plot(x,above_scatter_y,seriestype=:scatter,markerstrokecolor=:auto)
plot!(x,below_scatter_y,seriestype=:scatter,markerstrokecolor=:auto)
plot!(x,y,linewidth=2,linecolor=:red)

savefig("3-binary_linear_classification\\jlbinary_linear_classification.png")
display(Plots.plot!())
wait()
exit()

File: 3-binary_linear_classification/binary_linear_classification.py
import matplotlib.pyplot as pyplot
import numpy as np
import os
from datetime import datetime

m = -1
b = 3

x = np.linspace(0,10,100)
y = m*x+b

above_scatter_y = m*x+b+(abs(2*np.random.randn(100)))*x
below_scatter_y = m*x+b-(abs(2*np.random.randn(100)))*x

pyplot.scatter(x,above_scatter_y)
pyplot.scatter(x,below_scatter_y)
pyplot.plot(x,y,color='red')
if os.name=="nt":
    pyplot.savefig(f"plot{datetime.timestamp(datetime.now())}.png")
else:
    pyplot.savefig(f"plot{datetime.timestamp(datetime.now())}.png")
pyplot.show()
